src/templates/finops-hub/modules/storage.bicep

```
@@ -76,9 +90,181 @@ resource storageAccount 'Microsoft.Storage/storageAccounts@2022-09-01' = {
   tags: union(tags, contains(tagsByResource, 'Microsoft.Storage/storageAccounts') ? tagsByResource['Microsoft.Storage/storageAccounts'] : {})
   properties: {
     supportsHttpsTrafficOnly: true
+    allowSharedKeyAccess: true
     isHnsEnabled: true
     minimumTlsVersion: 'TLS1_2'
     allowBlobPublicAccess: false
+    publicNetworkAccess: 'Enabled'
```

Should this be based on enablePublicAccess?
```suggestion
    publicNetworkAccess: enablePublicAccess ? 'Enabled' : 'Disabled'
```

---

We need to keep this enabled so that azure services on the trusted list (like cost management exports) can access the storage account behind the firewall.
This is the middle radio button option on the storage account firewall.